### PR TITLE
US028: handle comma separated values for parameter

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by JakubW-k — APPROVED** [review]

File: src/main/java/org/example/daos/JobRoleDao.java [hunk 1]
```diff
@@ -52,59 +53,72 @@ private static String baseQuery() {
...
+            } else {
+                throw new IllegalArgumentException(
+                        "Unsupported parameter type: " + param.getClass().getName());
+            }
+        }
+    }
+
+    private void appendFilter(
+            final StringBuilder query, final List<Object> parameters, final String key, final Object value) {
+        if (value != null && !value.toString().isBlank()) {
```
Comment (Alicja-Kainos) [inline src/main/java/org/example/daos/JobRoleDao.java]: refactor this, add else

**Review by Alicja-Kainos — APPROVED** [review]

File: src/main/java/org/example/daos/JobRoleDao.java [hunk 2]
```diff
@@ -52,59 +53,100 @@ private static String baseQuery() {
...
-            query.append(" AND roleName LIKE ?");
-            parameters.add(jobRequest.getLikeRoleName());
+    private void executeFilteredJobQuery(
+            final StringBuilder query, final List<Object> parameters, final List<JobRole> jobRoles)
+            throws SQLException, ResultSetException {
+        try (Connection connection = DatabaseConnector.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query.toString())) {
+
+            setParameters(statement, parameters);
+            System.out.println(statement);
```
Comment (amordzon) [inline src/main/java/org/example/daos/JobRoleDao.java:65]: maybe remove this line

File: src/main/java/org/example/models/JobRoleFilteredRequest.java [hunk 3]
```diff
@@ -31,34 +30,34 @@ public String getRoleName() {
     }
 
     public String getLikeRoleName() {
-        return "%" + roleName + "%";
+        return roleName;
```
Comment (amordzon) [inline src/main/java/org/example/models/JobRoleFilteredRequest.java:36]: %%

**Review by amordzon — APPROVED** [review]
nice

File: src/main/java/org/example/daos/JobRoleDao.java (hunk 2 above)
Comment (amordzon) [reply src/main/java/org/example/daos/JobRoleDao.java:65]: i mean system out println

**Pull request merged**